src/safeposix/syscalls/fs_calls.rs

```diff
@@ -2261,107 +2261,188 @@ impl Cage {
+    /// Some of the examples of such bits are `S_IRUSR` (read by owner), `S_IWUSR` (write by owner), etc.
+    ///
+    /// ### Returns
+    /// 
+    /// Upon successful completion, zero is returned.
+    /// In case of a failure, -1 is returned, and `errno` is set depending on the error, e.g. EACCES, ENOENT, etc.
+    ///
+    /// ### Errors and Panics
+    ///
+    /// Currently, only two errors are supposrted:
+    /// * `EINVAL` - the value of the mode argument is invalid 
+    /// * `ENOENT` - a component of path does not name an existing file
+    /// Other errors, like `EFAULT  , `ENOTDIR`, etc. are not supported.
+    ///
+    /// There are no cases where this syscall panics.
+    ///
+    /// To learn more about the syscall, valid mode bits, and error values, see
+    /// [chmod(2)](https://man7.org/linux/man-pages/man2/chmod.2.html)
+
+    pub fn _chmod_helper(inodenum: usize, mode: u32) -> i32 {
+        //S_IRWXA is a result of bitwise-or'ing read, write, and execute or search permissions for the file owner, group owners, 
+        //and other users. It encompasses all the mode bits that can be changed via `chmod_syscall()` and is used as a bitmask
+        //to make sure that no other invalid bit change is being made. 
+        if (mode & S_IRWXA) == mode {
+            //getting a mutable reference to an inode struct that corresponds to the file whose mode bits we want to change
+            let mut thisinode = FS_METADATA.inodetable.get_mut(&inodenum).unwrap();
+            //log is used to store all the changes made to the filesystem. After the cage is closed, all the collected
+            //changes are serialized and the state of the underlying filsystem is persisted. This allows us to avoid
+            //serializing and persisting filesystem state after every `chmod_syscall()`.
+            let mut log = true;
+            //We obtain the mode bits that should remain intact by bitwise-and'ing the inode's mode bits with 
+            //the set of bits that can be changed via `chmod_syscall`. The changes are applied by bitwise-or'ing 
+            //the intact mode bits with the changed mode bits.
             match *thisinode {
                 Inode::File(ref mut general_inode) => {
-                    general_inode.mode = (general_inode.mode & !S_IRWXA) | mode
+                    general_inode.mode = (general_inode.mode & !S_IRWXA) | mode;
                 }
                 Inode::CharDev(ref mut dev_inode) => {
                     dev_inode.mode = (dev_inode.mode & !S_IRWXA) | mode;
                 }
                 Inode::Socket(ref mut sock_inode) => {
                     sock_inode.mode = (sock_inode.mode & !S_IRWXA) | mode;
+                    //Sockets only exist as long as the cages using them are running. After these cages are closed, no changes
+                    //to sockets' inodes need to be persisted, thus using log is unnecessary.
                     log = false;
```

Comment:
If we are saying that no changes to `sockets` need to be persisted, then why are we updating the mode for Socket type inodes?  We can simply ignore them as well, as we are not updating the metadata.

Reply:
"no changes need to be persisted" means that after the cage is closed, we don't need to know anything about its sockets and what happened to them. However, while the cage is still running, the changes need to be updated.